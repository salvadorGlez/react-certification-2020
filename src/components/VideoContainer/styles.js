import styled from 'styled-components';

export const Video = styled.div`
    @keyframes showTooltip {
        from { opacity: 0; }
        to { opacity: 1; }
    }
    margin: 15px;
    width: 320px;
    box-shadow: 0px 1px 3px -1px #000;
    border-radius: 4px;

    .img__container {
        width: 320px;
        height: 180px;
        position: relative;
        overflow: hidden;
        border-radius: 5px 5px 0px 0px;
        :hover {

            .img {
                filter: grayscale(100%) blur(4px) brightness(50%);
            }

            .video__desc {
                //visibility: visible;
                opacity: 1;
            }
        }
        
        .img {
            transition: all 100ms;
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
            height: 100%;
            background-image: url('${props => props.img}');
            background-repeat: no-repeat;
            background-size: cover;
            background-position: center;
            z-index: 1;
            backface-visibility: hidden;
        }

        .video__desc {
            z-index: 2;
            //visibility: hidden;
            transition: 250ms;
            position: relative;
            padding: 25px 20px;
            font-size: 16px;
            line-height: 18px;
            color: #fff;
            opacity: 0;
        }
    }

    .video__title {
        position: relative;
        margin: 5px 0px;
        padding: 0px 10px;
        :hover {
            .tooltip__text {
                display: inline-block;
            }
        }

        .title {
            white-space: nowrap;
            overflow: hidden;
            text-overflow: ellipsis;
            color: #424242;
        }

        .tooltip__text {
            padding: 5px 10px;
            border-radius: 5px;
            opacity: 0;
            position: absolute;
            display: none;
            background-color: #424242;
            color: #fff;
            top: 10px;
            left: 10px;
            animation-name: showTooltip;
            animation-duration: .2s;
            //animation-delay: 1s;
            animation-fill-mode: forwards;
        }
    }

    .video__channel {
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
        padding: 0px 10px;
        margin: 5px 0px;
        
        a {
            color: #424242;
            text-decoration: none;
        }
    }
`;