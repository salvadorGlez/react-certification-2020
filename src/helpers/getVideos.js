
export const getVideos = ({ items }) => {
    
    const videos = items
        .filter(item => item.id.kind.includes('video'))
        .map(({ id, snippet }) => ({
            id: id.videoId,
            title: snippet.title,
            description: snippet.description,
            channel: {
                id: snippet.channelId,
                title: snippet.channelTitle,
            },
            thumbnailUrl: snippet.thumbnails.medium.url
        }));

    return videos;

}
