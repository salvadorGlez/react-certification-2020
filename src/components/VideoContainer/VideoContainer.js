import React, { useRef } from 'react'
import { Video } from './styles';

export const VideoContainer = ({ video }) => {
    const spanRef = useRef();

    const {
        // id,
        title,
        description,
        channel,
        thumbnailUrl
    } = video;

    return (
        <Video img={ thumbnailUrl }>
            <div className="img__container">
                <div className="img"></div>
                <p className="video__desc">{ description }</p>
            </div>
            <div className="video__title">
                <h3 className="title">
                    { title }
                </h3>
                <span
                    ref={ spanRef }
                    className="tooltip__text"
                >
                    { title }
                </span>
            </div>
            <div className="video__channel">
                <a
                    href={ `https://www.youtube.com/channel/${ channel.id }` }
                    target="_blank"
                    rel="noreferrer"
                >
                    { channel.title }
                </a>
            </div>
        </Video>
    )
}
