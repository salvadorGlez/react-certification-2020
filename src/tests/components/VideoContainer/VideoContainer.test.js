import { shallow } from 'enzyme';
import { VideoContainer } from '../../../components/VideoContainer/VideoContainer';
import { getVideos } from '../../../helpers/getVideos';

describe('Tests in <VideoContainer />', () => {
    const ytData = require('../../../data/videos.json');
    const [video] = getVideos(ytData);
    const wrapper = shallow(<VideoContainer video={ video } />);
   
    test('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should the video title be displayed correctly', () => {
        expect(wrapper.find('span').text()).toBe(video.title);
    });
    
    test('should the video channel be a link to the channels page and be displayed correctly', () => {
        const a = wrapper.find('a');
        expect(a.text()).toBe(video.channel.title);
        expect(a.prop('href')).toBe(`https://www.youtube.com/channel/${ video.channel.id }`);
    });
    
});
