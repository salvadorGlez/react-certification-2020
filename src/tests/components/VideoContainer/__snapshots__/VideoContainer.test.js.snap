// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <VideoContainer /> should match snapshot 1`] = `
<styled.div
  img="https://i.ytimg.com/vi/nmXMgqjQzls/mqdefault.jpg"
>
  <div
    className="img__container"
  >
    <div
      className="img"
    />
    <p
      className="video__desc"
    >
      Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office. In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, ...
    </p>
  </div>
  <div
    className="video__title"
  >
    <h3
      className="title"
    >
      Video Tour | Welcome to Wizeline Guadalajara
    </h3>
    <span
      className="tooltip__text"
    >
      Video Tour | Welcome to Wizeline Guadalajara
    </span>
  </div>
  <div
    className="video__channel"
  >
    <a
      href="https://www.youtube.com/channel/UCPGzT4wecuWM0BH9mPiulXg"
      rel="noreferrer"
      target="_blank"
    >
      Wizeline
    </a>
  </div>
</styled.div>
`;
